api/tests/app/test_app.py

```diff
@@ -0,0 +1,162 @@
+import pytest
+
+from api.models.app import AppDTO
+from api.tests.utils.common_functions import get_app_from_ini_config_file_json, json_multi_type_to_json_string, \
+    create_app_config, camel_case_to_snake_case_dict
+# The line below is absolutely necessary. Fixtures are passed as arguments to test functions. That is why IDE could
+# not recognized them.
+from api.tests.utils.fixtures_tests import config_rollback, app_config
+
+
+def expected_response_function(config_sample_path):
+    expected_response = get_app_from_ini_config_file_json(config_sample_path)
+    expected_response = json_multi_type_to_json_string(expected_response)
+
+    # It is weird that this change does not appear in response.json()
+    # Because if request was successfully, the PUT request will change the flag "has_been_configured"
+    expected_response["has_been_configured"] = "False"
+
+    return expected_response
+
+
+def expected_response_default():
+    """When no valid key is provided to the PUT request, config is updated with default values stored in AppDTO"""
+    expected_response = AppDTO().__dict__
+    expected_response = camel_case_to_snake_case_dict(expected_response)
+    expected_response = json_multi_type_to_json_string(expected_response)
+    expected_response["dashboard_url"] = expected_response["dashboard_u_r_l"]
```

Comment:
I think that this line should be updated in the fuction `camel_case_to_snake_case_dict`.

Reply:
I decided to include that line in the test instead of the function camel_case_to_snake_case_dict, to maintain consistency between the function name and its functionality. However, i can change it if you insist, just want you to know why i did it.

Reply:
Mmmm, maybe I am wrong, but I think the snake_case of `dashboardURL` is `dashboard_url`. URL is an acronym, and in snake case we usually don't split them with `_` (for example API, csv, etc).
I think that is ok include this "border" case in the function camel_case_to_snake_case_dict.